16_wordle/main.go

```diff
@@ -87,15 +88,80 @@ func (l List) Random() []byte {
 	- Return slice of contained characters
 */
 
-func checkChar(chosenWordSlice []string, inputSlice []string) []string {
-	var containedSlice []string
+// func checkChar(chosenWordSlice []string, inputSlice []string) []string {
+// 	var containedSlice []string
+
+// 	for _, value := range inputSlice {
+// 		// debug
+// 		// fmt.Println("value print", value)
+// 		if slices.Contains(chosenWordSlice, value) {
+// 			containedSlice = append(containedSlice, value)
+// 		}
+// 	}
+// 	return containedSlice
+// }
+
+func checkCharPrecise(chosenWordSlice []string, inputSlice []string) []int {
+	var containedSlice []int
+	// 1 = contained but incorrect loc
+	// 2 = contained in correc loc
+	// 0 not contained
+
+	// correct word check
+	if slices.Equal(chosenWordSlice, inputSlice) {
+		return []int{2, 2, 2, 2, 2}
+	}
 
-	for _, value := range inputSlice {
-		// debug
-		// fmt.Println("value print", value)
-		if slices.Contains(chosenWordSlice, value) {
-			containedSlice = append(containedSlice, value)
-		}
+	// WIP refactor
+	// for index, value := range chosenWordSlice {
+	// 	if value == chosenWordSlice[index] {
+	// 		containedSlice = append(containedSlice, 2)
+	// 	} else if slices.Contains(chosenWordSlice, inputSlice[index]) {
+	// 		containedSlice = append(containedSlice, 1)
+	// 	} else {
+	// 		containedSlice = append(containedSlice, 0)
+	// 	}
+	// }
+
+	if inputSlice[0] == chosenWordSlice[0] {
+		containedSlice = append(containedSlice, 2)
+	} else if slices.Contains(chosenWordSlice, inputSlice[0]) {
+		containedSlice = append(containedSlice, 1)
+	} else {
+		containedSlice = append(containedSlice, 0)
+	}
```

Comment:
this is actually the logic I would have implemented, can you use a loop now instead of copying / pasting this code many times?